Allow @opentelemetry@1.9.0

maisano-patreon opened the pull request:
### What does this PR do?

Allows for `@opentelemetry/api@1.9.0`.

### Motivation

A tracing regression due to using the latest version of `@opentelemetry/api`.

### Plugin Checklist
<!-- Fill this section if adding or updating a plugin. Remove otherwise. -->

- [ ] Unit tests.
- [ ] TypeScript [definitions][1].
- [ ] TypeScript [tests][2].
- [ ] API [documentation][3].
- [ ] CircleCI [jobs/workflows][4].
- [ ] Plugin is [exported][5].

[1]: https://github.com/DataDog/dd-trace-js/blob/master/index.d.ts
[2]: https://github.com/DataDog/dd-trace-js/blob/master/docs/test.ts
[3]: https://github.com/DataDog/documentation/blob/master/content/en/tracing/trace_collection/library_config/nodejs.md
[4]: https://github.com/DataDog/dd-trace-js/blob/master/.circleci/config.yml
[5]: https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/index.js

### Additional Notes

Happy to make whatever changes to land this.

BridgeAR commented on package.json:
```diff
@@ -92,7 +92,7 @@
     "@datadog/sketches-js": "^2.1.0",
     "@isaacs/ttlcache": "^1.4.1",
-    "@opentelemetry/api": ">=1.0.0 <1.9.0",
+    "@opentelemetry/api": ">=1.0.0 <=1.9.0",
```
```suggestion
    "@opentelemetry/api": ">=1.0.0 <1.10.0",
```

BridgeAR reviewed (commented):
@maisano-patreon thank you for the PR. We still have a reference about the version in our main README.md. Can you please also update that?
I would actually update the text not to include any specific version range so that we do not have to update it every time.

I am not certain if we have to test for anything in particular/update something to make sure the instrumentation works for the new version as well.

maisano-patreon commented:
@BridgeAR Thanks for flagging the README reference. I removed the version specifier altogether and changed the language to be more broad. Also updated the `package.json` per your recommendation.

BridgeAR reviewed (approved):
LGTM with the CI being green. Sadly, we can't currently run the CI for external contributors, so I'll later on cherry-pick the commit and open a new PR with this.

BridgeAR commented:
@maisano-patreon the tests are failing with this and I can't fix those right now. Could you have a look at these failures? I would just port the fixes over as well. See https://github.com/DataDog/dd-trace-js/pull/5583#issuecomment-2809949128
